scapy/utils.py

```diff
@@ -1905,6 +1905,250 @@ def write_packet(self,
         )
 
 
+@conf.commands.register
+def rderf(filename, count=-1):
+    # type: (Union[IO[bytes], str], int) -> PacketList
+    """Read a ERF file and return a packet list
+
+    :param count: read only <count> packets
+    """
+    with ERFEthernetReader(filename) as fdesc:
+        return fdesc.read_all(count=count)
+
+
+class ERFEthernetReader(_SuperSocket):
+
+    def __init__(self, filename):  # type: ignore
+        # type: (str) -> None
+        self.filename, self.f = self.open(filename)
+        self.power = Decimal(10) ** Decimal(-9)
+
+    # time is in 64-bits Endace's format which can be see here:
+    # https://www.endace.com/erf-extensible-record-format-types.pdf
+    def _convert_erf_timestamp(self, t):
+        # type (int) -> EDecimal
+        sec = t >> 32
+        frac_sec = t & 0xffffffff
+        frac_sec *= 10**9
+        frac_sec += (frac_sec & 0x80000000) << 1
+        frac_sec >>= 32
+        return EDecimal(sec + self.power * frac_sec)
+
+    def __enter__(self):
+        # type: () -> ERFEthernetReader
+        return self
+
+    def __exit__(self, exc_type, exc_value, tracback):
+        # type: (Optional[Any], Optional[Any], Optional[Any]) -> None
+        self.f.close()
+
+    @staticmethod
+    def open(fname  # type: Union[IO[bytes], str]
+             ):
+        # type: (...) -> Tuple[str, _ByteStream, bytes]
+        """Open (if necessary) filename"""
+        if isinstance(fname, str):
+            filename = fname
+            if filename.endswith(".gz"):
```

Comment:
We did this because there is no file header in ERF, unlike PCAP where we can read the first 4 bytes (magic number). So the exception is only thrown when you read bytes from files that are not gzip but opened as gzip. 
Another option to do this is to try to read the file anyway, check for error, and close and reopen the file again to read from the beginning. Do you think the 2nd option is better? or do you have another solution in mind?

Reply:
I will prefer the second option. That way this will be work simlarly as pcap files.